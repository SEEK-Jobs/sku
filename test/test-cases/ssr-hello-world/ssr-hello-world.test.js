const path = require('path');

const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const gracefulSpawn = require('../../../lib/gracefulSpawn');
const waitForUrls = require('../../utils/waitForUrls');
const getAppSnapshot = require('../../utils/getAppSnapshot');
const startAssetServer = require('../../utils/assetServer');
const skuBuildConfig = require('./sku-build.config');
const skuStartConfig = require('./sku-start.config');

const getTestConfig = skuConfig => ({
  backendUrl: `http://localhost:${skuConfig.serverPort}`,
  targetDirectory: path.join(__dirname, skuConfig.target)
});

describe('ssr-hello-world', () => {
  describe('start', () => {
    const { backendUrl } = getTestConfig(skuStartConfig);
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('start-ssr', __dirname, [
        '--config=sku-start.config.js'
      ]);
      await waitForUrls(backendUrl);
    });

    afterAll(() => {
      server.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(backendUrl);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('build', () => {
    const { backendUrl, targetDirectory } = getTestConfig(skuBuildConfig);
    let server, closeAssetServer;

    beforeAll(async () => {
      await runSkuScriptInDir('build-ssr', __dirname, [
        '--config=sku-build.config.js'
      ]);

      server = gracefulSpawn('node', ['server'], {
        cwd: targetDirectory,
        stdio: 'inherit'
      });
      closeAssetServer = startAssetServer(4000, targetDirectory);

      await waitForUrls(backendUrl, 'http://localhost:4000');
    });

    afterAll(() => {
      server.kill();
      closeAssetServer();
    });

    it('should generate a production server', async () => {
      const snapshot = await getAppSnapshot(backendUrl);
      expect(snapshot).toMatchSnapshot();
    });
  });
});
