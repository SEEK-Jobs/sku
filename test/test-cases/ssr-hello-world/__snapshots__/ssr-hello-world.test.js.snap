// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr-hello-world build should generate a production server 1`] = `
Object {
  "content": "<!DOCTYPE html><html><head>
      <meta charset=\\"UTF-8\\">
      <title>hello-world</title>
      <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://localhost:4000/main-6684d202b10a3b9bde2c.css\\">
    </head>
    <body>
      <div id=\\"app\\"><div class=\\"_1brbpTo Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"_3YtmnHj lwsthgY\\">Hello World</div></div></div>
      <script type=\\"text/javascript\\" src=\\"http://localhost:4000/runtime-46e490315fd43280bd3a.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4000/vendors~main-7433479861943aaac3dc.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4000/main-7617f4b40ec6550cdb75.js\\" crossorigin=\\"anonymous\\"></script>
    
  
</body></html>",
  "errors": Array [],
  "sourceHtml": "
  <!DOCTYPE html>
  <html>
    <head>
      <meta charset=\\"UTF-8\\">
      <title>hello-world</title>
      <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://localhost:4000/main-6684d202b10a3b9bde2c.css\\" />
    </head>
    <body>
      <div id=\\"app\\"><div class=\\"_1brbpTo Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"_3YtmnHj lwsthgY\\">Hello World</div></div></div>
      <script type=\\"text/javascript\\" src=\\"http://localhost:4000/runtime-46e490315fd43280bd3a.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4000/vendors~main-7433479861943aaac3dc.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4000/main-7617f4b40ec6550cdb75.js\\" crossorigin=\\"anonymous\\"></script>
    </body>
  </html>
",
  "warnings": Array [],
}
`;

exports[`ssr-hello-world start should start a development server 1`] = `
Object {
  "content": "<!DOCTYPE html><html><head>
      <meta charset=\\"UTF-8\\">
      <title>hello-world</title>
      <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://localhost:8100/main.css\\">
    </head>
    <body>
      <div id=\\"app\\"><div class=\\"lessStyles__root___1brbpTo jsStyles-css__root___Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"lessStyles__nested___3YtmnHj jsStyles-css__nested___lwsthgY\\">Hello World</div></div></div>
      <script type=\\"text/javascript\\" src=\\"http://localhost:8100/runtime.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:8100/vendors~main.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:8100/main.js\\" crossorigin=\\"anonymous\\"></script>
    
  
</body></html>",
  "errors": Array [],
  "sourceHtml": "
  <!DOCTYPE html>
  <html>
    <head>
      <meta charset=\\"UTF-8\\">
      <title>hello-world</title>
      <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://localhost:8100/main.css\\" />
    </head>
    <body>
      <div id=\\"app\\"><div class=\\"lessStyles__root___1brbpTo jsStyles-css__root___Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"lessStyles__nested___3YtmnHj jsStyles-css__nested___lwsthgY\\">Hello World</div></div></div>
      <script type=\\"text/javascript\\" src=\\"http://localhost:8100/runtime.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:8100/vendors~main.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:8100/main.js\\" crossorigin=\\"anonymous\\"></script>
    </body>
  </html>
",
  "warnings": Array [],
}
`;
